### DISCUSSION TITLE
Windows not allowing another session

### DISCUSSION THREAD

**Discussion by daddyodeals** [jdholtz/auto-southwest-check-in#88] [General]
Windows isn't allowing multiple sessions, unless I'm using it wrong.  I opened up another CMD prompt via Admin and the script gave me a "log already in use" permissions error.  I tried it on AWS (Windows instance) and a local PC (Windows instance) with the same results.

Traceback (most recent call last):
  File "c:\Python39\auto-southwest-check-in\southwest.py", line 12, in <module>
    main(arguments)
  File "c:\Python39\auto-southwest-check-in\lib\main.py", line 119, in main
    log.init_main_logging()
  File "c:\Python39\auto-southwest-check-in\lib\log.py", line 21, in init_main_logging
    logger.handlers[0].doRollover()  # Create a new log file when starting the application
  File "c:\Python39\lib\logging\handlers.py", line 177, in doRollover
    self.rotate(self.baseFilename, dfn)
  File "c:\Python39\lib\logging\handlers.py", line 115, in rotate
    os.rename(source, dest)
PermissionError: [WinError 32] The process cannot access the file because it is being used by another process: 'c:\\Python39\\auto-southwest-check-in\\logs\\auto-southwest-check-in.log' -> 'c:\\Python39\\auto-southwest-check-in\\logs\\auto-southwest-check-in.log.1'

**Comment by jdholtz** [comment]
Due to the log system, I don’t think it will work well with two different sessions. However, you shouldn’t need to have different sessions as you can add multiple accounts and flights to the same script via the configuration file. Let me know if there is another reason you want to run multiple different instances. 

Additionally, to run multiple instances you can also use different Docker containers.

**Comment by daddyodeals** [comment]
Thanks for the quick reply.  The only reason I'd want to run multiple different instances is if I don't have all my check ins on hand and have to add another one later.  From what you're saying, I guess I'd just stop my running instance, update the config, then re-run it?

**Reply by jdholtz** [reply]
Yes. Another option (if the credentials are accessible to you) is to use accounts and new check-ins will automatically be scheduled as the flights are scheduled for those accounts. It just might take a day until you get the notification that it was scheduled if you have it check once a day (the default).

**Comment by daddyodeals** [comment]
Again, thanks for the quick answers.  

Will the script keep running for the check in on my inbound flight, too (same PNR), or do I need to re-run it?

**Reply by jdholtz** [reply]
Yes, it will run the check-in for all flights on a reservation.
